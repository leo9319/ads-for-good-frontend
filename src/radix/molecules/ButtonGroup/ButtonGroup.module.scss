.buttonGroupContainer {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  gap: 16px;
}

.buttonWrapper {
  flex: 1;
  display: flex;
  justify-content: center;
  box-sizing: border-box;
}

.buttonWrapper button {
  width: 100%;
}

.iconOnly {
  flex: 0;
  width: auto;
  padding: 0;

  button {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: var(--spacing-xs);
  }
}
