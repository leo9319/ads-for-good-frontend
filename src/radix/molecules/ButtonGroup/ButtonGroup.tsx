import React, { ReactElement } from 'react';
import Button, { ButtonProps } from '@radix-styles/atoms/Button';
import Flex from '@radix-styles/atoms/Flex';
import Box from '@radix-styles/atoms/Box';
import { classNames } from '@utils/common/classNames';

import styles from './ButtonGroup.module.scss';

export interface ButtonGroupProps {
  /**
   * List of buttons to display in the group
   * @default []
   * @property
   */
  data: Array<ButtonProps>;
  /**
   * Determines the layout of buttons
   * @default 'horizontal'
   * @property
   */
  direction?: string;
  /**
   * Custom class name for the button group
   */
  className?: string;
  /**
   * The Button will be disabled if it's true
   * @property
   */
  disabled?: boolean;
}

export const ButtonGroup = ({
  data = [],
  direction = 'horizontal',
  className,
  disabled,
}: ButtonGroupProps): ReactElement => {
  const isVertical = direction === 'vertical';
  const uniqueId = React.useId();
  const getUniqueKey = (index: number) =>
    'button-wrapper-' + index + '-' + uniqueId;

  return (
    <Flex
      className={classNames(styles.buttonGroupContainer, className)}
      direction={isVertical ? 'column' : 'row'}
      display="flex"
      gap="3"
    >
      {data?.map((props: ButtonProps, index: number) => {
        const isDisabled = disabled ?? props.disabled;
        const isIconOnly = !props.text && !!props.icon;
        const classes = classNames(
          styles.buttonWrapper,
          isIconOnly ? styles.iconOnly : undefined
        );
        return (
          <Box asChild key={getUniqueKey(index)} className={classes}>
            <Button {...props} disabled={isDisabled} />
          </Box>
        );
      })}
    </Flex>
  );
};

export default ButtonGroup;
