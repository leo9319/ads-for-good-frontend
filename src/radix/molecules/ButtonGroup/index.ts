export type { ButtonGroupProps } from './ButtonGroup';
export { default } from './ButtonGroup';
export { ButtonGroup } from './ButtonGroup';
