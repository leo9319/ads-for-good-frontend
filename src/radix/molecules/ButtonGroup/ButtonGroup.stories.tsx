import React from 'react';
import ButtonGroup from './ButtonGroup';
import { Meta, StoryObj } from '@storybook/react/*';

const meta = {
  title: 'Library/Molecules/ButtonGroup',
  component: ButtonGroup,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    data: {
      description: 'Button Group Data (Array of Button Objects)',
      control: {
        type: 'object',
      },
    },
    direction: {
      description: 'Button Group Direction (Horizontal/Vertical)',
      options: ['horizontal', 'vertical'],
      control: {
        type: 'select',
      },
    },
    disabled: {
      description: 'To disable the button',
      options: [true, false],
      control: { type: 'boolean' },
    },
  },
  args: {
    direction: 'horizontal',
  },
  decorators: [
    Story => (
      <div style={{ padding: '20px' }}>
        <Story />
      </div>
    ),
  ],
} satisfies Meta<typeof ButtonGroup>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Button: Story = {
  args: {
    data: [
      {
        text: 'Button 1',
      },
    ],
  },
};

export const ButtonPair: Story = {
  args: {
    data: [
      {
        text: 'Button 1',
        icon: 'icon-plus',
      },
      {
        text: 'Button 2',
        icon: 'icon-arrow',
        iconLeft: true,
      },
    ],
  },
};

export const IconPlusButton: Story = {
  args: {
    data: [
      {
        icon: 'icon-chevron-left',
        iconLeft: true,
        size: 'lg',
        mode: 'outline',
      },
      {
        text: 'Button 1',
      },
    ],
  },
};

export const IconButtonGroup: Story = {
  args: {
    data: [
      {
        text: 'Button 1',
        icon: 'icon-plus',
      },
      {
        text: 'Button 2',
        icon: 'icon-arrow',
        iconLeft: true,
      },
    ],
  },
};

export const TripleButton: Story = {
  args: {
    data: [
      {
        text: 'Button 1',
      },
      {
        text: 'Button 2',
      },
      {
        text: 'Button 3',
      },
    ],
  },
};

export const Disabled: Story = {
  args: {
    data: [
      {
        text: 'Button',
      },
    ],
    disabled: true,
  },
};
