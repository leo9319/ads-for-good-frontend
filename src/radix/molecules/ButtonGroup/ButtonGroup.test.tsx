import React from 'react';
import ButtonGroup from './ButtonGroup';
import { render } from '@testing-library/react';

describe('ButtonGroup', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <ButtonGroup
        data={[
          { text: 'Button ', icon: '+' },
          { icon: '+' },
          { text: 'Only Button' },
        ]}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
