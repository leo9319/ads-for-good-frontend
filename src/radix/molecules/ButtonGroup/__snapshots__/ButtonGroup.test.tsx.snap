// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Flex rt-r-display-flex rt-r-fd-row rt-r-gap-3 buttonGroupContainer"
    >
      <button
        aria-label="Button "
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton xl primary none rt-Box buttonWrapper"
        data-accent-color=""
      >
        <div
          class="buttonContent iconLeft"
        >
          <span>
            Button 
          </span>
          <span
            class="icon"
            style="border-radius: 0; display: grid; align-items: center; justify-content: center;"
          >
            <span
              class="icon +"
            />
          </span>
        </div>
      </button>
      <button
        aria-label="+"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton xl primary none rt-Box buttonWrapper iconOnly"
        data-accent-color=""
      >
        <div
          class="buttonContent iconLeft iconOnly"
        >
          <span
            class="icon"
            style="border-radius: 0; display: grid; align-items: center; justify-content: center;"
          >
            <span
              class="icon +"
            />
          </span>
        </div>
      </button>
      <button
        aria-label="Only Button"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton xl primary none rt-Box buttonWrapper"
        data-accent-color=""
      >
        <div
          class="buttonContent textOnly"
        >
          <span>
            Only Button
          </span>
        </div>
      </button>
    </div>
  </div>
</body>
`;
